### PR TITLE
[TestCase]=>column lineage not track for recursive cols

### PR DESCRIPTION
**Pull request by hloombaupgrade**
### Problem
With the query in test case desired 

```
[
            ColumnLineage {
                descendant: ColumnMeta {
                    origin: None,
                    name: "col_1".to_string()
                },
                lineage: vec![ColumnMeta {
                    origin: Some(table("**_source_tbl_**")),
                    name: "col_1".to_string()
                }]
            },
            ColumnLineage {
                descendant: ColumnMeta {
                    origin: None,
                    name: "col_3".to_string()
                },
                lineage: vec![ColumnMeta {
                    origin: Some(table("**_source_tbl_**")),
                    name: "col_3".to_string()
                }]
            },
            ColumnLineage {
                descendant: ColumnMeta {
                    origin: None,
                    name: "col_4".to_string()
                },
                lineage: vec![ColumnMeta {
                    origin: Some(table("**_source_tbl_**")),
                    name: "col_4".to_string()
                }]
            },
            ColumnLineage {
                descendant: ColumnMeta {
                    origin: None,
                    name: "x".to_string()
                },
                lineage: vec![ColumnMeta {
                    origin: Some(table("tbl2")),
                    name: "x".to_string()
                }]
            },
            ColumnLineage {
                descendant: ColumnMeta {
                    origin: None,
                    name: "y".to_string()
                },
                lineage: vec![ColumnMeta {
                    origin: Some(table("tbl3")),
                    name: "y".to_string()
                }]
            },
        ]
```

Closes: #ISSUE-NUMBER

### Solution

Please describe your change as it relates to the problem, or bug fix, as well as any dependencies. If your change requires a schema change, please describe the schema modification(s) and whether it's a _backwards-incompatible_ or _backwards-compatible_ change, then select one of the following:

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [ ] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [ ] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [ ] Your changes are accompanied by tests (_if relevant_)
- [ ] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] Your comment includes a one-liner for the changelog about the specific purpose of the change (_not required for changes to tests, docs, or CI config_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2024 contributors to the OpenLineage project

### CODE REVIEW

File: integration/sql/impl/tests/column_lineage/tests_cte.rs [hunk 1]
```diff
@@ -39,3 +39,80 @@ fn test_simple_cte() {
...
+    assert_eq!(
+        output.column_lineage,
+        vec![
+            ColumnLineage {
+                descendant: ColumnMeta {
+                    origin: None,
+                    name: "col_1".to_string()
+                },
+                lineage: vec![ColumnMeta {
+                    origin: Some(table("stage_1")),
```
Comment (hloombaupgrade) [inline integration/sql/impl/tests/column_lineage/tests_cte.rs:72]: this should be source_tbl

File: integration/sql/impl/tests/column_lineage/tests_cte.rs [hunk 2]
```diff
@@ -39,3 +39,80 @@ fn test_simple_cte() {
...
+                    name: "col_1".to_string()
+                }]
+            },
+            ColumnLineage {
+                descendant: ColumnMeta {
+                    origin: None,
+                    name: "col_3".to_string()
+                },
+                lineage: vec![ColumnMeta {
+                    origin: Some(table("stage_1")),
```
Comment (hloombaupgrade) [inline integration/sql/impl/tests/column_lineage/tests_cte.rs:82]: this should be source_tbl

File: integration/sql/impl/tests/column_lineage/tests_cte.rs [hunk 3]
```diff
@@ -39,3 +39,80 @@ fn test_simple_cte() {
...
+                    name: "col_3".to_string()
+                }]
+            },
+            ColumnLineage {
+                descendant: ColumnMeta {
+                    origin: None,
+                    name: "col_4".to_string()
+                },
+                lineage: vec![ColumnMeta {
+                    origin: Some(table("stage_1")),
```
Comment (hloombaupgrade) [inline integration/sql/impl/tests/column_lineage/tests_cte.rs:92]: this should be source_tbl

**Comment by JDarDagran** [comment]
Closing this. https://github.com/OpenLineage/OpenLineage/pull/3107 fixes the issue.

**Pull request closed without merging**